managed-token/program/tests/test.rs

```diff
@@ -141,3 +172,116 @@ async fn test_spl_managed_token_basic() {
     .await
     .unwrap();
 }
+
+#[tokio::test]
+async fn test_spl_managed_token_with_delegate() {
+    let mut context = spl_managed_token_test().start_with_context().await;
+    let lwc = &mut context.banks_client;
+    let authority = Keypair::new();
+    transfer(lwc, &context.payer, &authority.pubkey(), sol(10.0))
+        .await
+        .unwrap();
+    let mint = Keypair::new();
+    let mint_key = mint.pubkey();
+    let create_ix =
+        create_initialize_mint_instruction(&mint_key, &authority.pubkey(), &authority.pubkey(), 0)
+            .unwrap();
+    process_transaction(lwc, vec![create_ix], vec![&authority, &mint])
+        .await
+        .unwrap();
+
+    let alice = Keypair::new();
+    let alice_key = alice.pubkey();
+    let bob = Keypair::new();
+    let bob_key = bob.pubkey();
+    let eve = Keypair::new();
+    let eve_key = eve.pubkey();
+
+    transfer(lwc, &context.payer, &alice_key, sol(1.0))
+        .await
+        .unwrap();
+    transfer(lwc, &context.payer, &bob_key, sol(1.0))
+        .await
+        .unwrap();
+
+    let create_alice_ata_ix = create_initialize_account_instruction(
+        &mint_key,
+        &alice_key,
+        &authority.pubkey(),
+        &authority.pubkey(),
+    )
+    .unwrap();
+    let mint_to_ix =
+        create_mint_to_instruction(&mint_key, &alice_key, &authority.pubkey(), 1).unwrap();
+    let delegate_ix =
+        create_approve_instruction(&mint_key, &alice_key, &bob_key, &authority.pubkey(), 1)
+            .unwrap();
+    process_transaction(
+        lwc,
+        vec![create_alice_ata_ix, mint_to_ix, delegate_ix],
+        vec![&alice, &authority],
+    )
+    .await
+    .unwrap();
+
+    let create_eve_ata_ix = create_initialize_account_instruction(
+        &mint_key,
+        &eve_key,
+        &authority.pubkey(),
+        &authority.pubkey(),
+    )
+    .unwrap();
+    let successful_transfer_ix = create_transfer_with_delegate_instruction(
+        &alice_key,
+        &eve_key,
+        &bob_key,
+        &mint_key,
+        &authority.pubkey(),
+        1,
+    )
+    .unwrap();
+    process_transaction(
+        lwc,
+        vec![create_eve_ata_ix, successful_transfer_ix],
+        vec![&bob, &authority],
+    )
+    .await
+    .unwrap();
```

Comment:
Could you check to make sure that the transfer was successful and that the delegate was set?